code-name-taran/src/com/spaghettisoft/component/scrabble/WordValidation.java

```diff
@@ -0,0 +1,150 @@
+			return false;
+		}
+		boolean areEnoughLEtters = checkIfPlayerHasEnoughLetters(word, row, col, orientation, board, player);
+		if(!areEnoughLEtters) {
+			System.out.println("Your do not have all the necessary letters for the word you wanna play.");
+			return false;
+		}
+		return true;
+	}
+
+	private static boolean checkIfWordFitsInTheBoard(String word, int row, int col, char orientation) {
+		if(orientation == 'h' && (col + word.length()) > 16) {
+			return false;
+		} else if (orientation == 'v' && (row + word.length()) > 16) {
+			return false;
+		}
+		return true;
+	}
+
+	private static boolean checkIfWordFitsWellOnTheTiles(String word, int row, int col, char orientation, Board board) {
+		Tile[] tiles = tilesOnWhichWordShouldBePlaced(word, row, col, orientation, board);
+		if(board.getTiles()[7][7].getContent().charAt(1) == '@') {
+			int counter = 0;
+			for (int i = 0; i < tiles.length; i++) {
+				if(tiles[i].getContent().charAt(1) == '@') {
+					counter++;
+					break;
+				}
+			}
+			if (counter == 0) {
+				return false;
+			}
+		} else {
+			int counter = 0;
+			for (int i = 0; i < tiles.length; i++) {
+				if(Character.isLetter(tiles[i].getContent().charAt(1))) {
+					counter++;
+					char contentChar = tiles[i].getContent().charAt(1);
+					char wordChar = Constants.makeTheLowercaseBecomeUppercase(word.charAt(i));
+					if(contentChar != wordChar) {
+						return false;
+					}
+				}
+			}
+			if(counter == 0) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	private static boolean checkIfPlayerHasEnoughLetters(String word, int row, int col, char orientation, Board board, Player player) {
+		ArrayList<Stave> memoryStick = fillMemoryStick(word, row, col, orientation, board);
+		String playersLetters = Checks.takeLettersPlayerHasInHand(player);
+		boolean hasAllLetters = hasThePlayerAllTheNecessaryLetters(memoryStick, playersLetters);
+		return hasAllLetters;
+	}
+
+	private static boolean hasThePlayerAllTheNecessaryLetters(ArrayList<Stave> memoryStick, String playersLetters) {
+		String newStringPlayersLetters = Constants.copyStringToAnotherString(playersLetters);
+		for (int i = 0; i < memoryStick.size(); i++) {
+			char ch = memoryStick.get(i).getLetter();
+			boolean hasLetter = Checks.checkIfStringContainsLetter(newStringPlayersLetters, ch);
+			boolean hasWildcard = Checks.checkIfStringContainsLetter(newStringPlayersLetters, '?');
+
+			if(!hasLetter && !hasWildcard) {
+				return false;
+			}
+			if(hasLetter) {
+				String newString = trimmCharFromString(newStringPlayersLetters, ch);
+				newStringPlayersLetters = newString;
+			} else {
+				String newString = trimmCharFromString(newStringPlayersLetters, '?');
+				newStringPlayersLetters = newString;
+			}
+		}
+		return true;
+	}
+
+	private static Tile[] tilesOnWhichWordShouldBePlaced(String word, int row, int col, char orientation, Board board) {

```

Comment:
Remember to name methods after verbs. Something like "getWordTiles()" is better suited : )